docs/0001-knowledge-graph-and-telemetry-specification.md

```diff
@@ -0,0 +1,118 @@
+
+# Knowledge Graph and telemetry specification
+
+# Worker Node
+
+Terminology:
+
+- Capacity - the maximum allocatable units of resource on the kubernetes worker node.
+
+Energy consumption: converting joules_total (Counter) to milliwatts.
+Average using window of 5 minutes (300 seconds):
+
+- irate(kepler_node_platform_joules_total[300s]) / 300 = watts
+- irate(kepler_node_platform_joules_total[300s]) *1000 / 300 = milliwatts
+
+| Metric | Source | Synthetic Data Generator | Knowledge graph, value of "glc:hasDescription" attribute of glc:Measurement | Tradeoff Service, (JsonPath) |
+| --- | --- | --- | --- | --- |
+| Node energy index, milliwatt | Node resource:<br/> $.metadata.annotations.['glaciation-project.eu/metric/node-energy-index'] |  | node-energy-index | $.worker_nodes.energy_index |
+| Node CPU capacity max, cores | Node resource:<br/> $.status.allocatable.cpu | WN_CPU_MAX_CAPACITY | cpu-capacity-max | $.worker_nodes.resources.cpu.max |
+| Node memory capacity max, Mb | Node resource:<br/> $.status.allocatable.memory | WN_MEM_MAX_CAPACITY | ram-capacity-max | $.worker_nodes.resources.memory.max |
+| Node gpu capacity max, Flops | Not available yet | WN_GPU_MAX_CAPACITY | gpu-capacity-max | $.worker_nodes.resources.gpu.max |
```

Comment:
I think now we have GPU into our cluster so we can update this file in the source field

Reply:
Yes, we can provide the metric. I'd say it is a separate work item requiring investigation.